server/src/main/java/org/opensearch/cluster/routing/allocation/AwarenessReplicaBalance.java

```diff
@@ -31,19 +33,55 @@
  * Helps in balancing shards across all awareness attributes and ensuring high availability of data.
  */
 public class AwarenessReplicaBalance {
+    public static final String SETTING_CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE = "cluster.routing.allocation.awareness.balance";
+    public static final String SETTING_USE_FORCE_ZONE_FOR_REPLICA = "cluster.use_force_zone_for_replica";
     public static final Setting<Boolean> CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING = Setting.boolSetting(
-        "cluster.routing.allocation.awareness.balance",
+        SETTING_CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE,
         false,
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
+    public static final Setting<Boolean> USE_FORCE_ZONE_FOR_REPLICA_SETTING = Setting.boolSetting(
+        SETTING_USE_FORCE_ZONE_FOR_REPLICA,
+        false,
+        new Setting.Validator<>() {
+
+            @Override
+            public void validate(final Boolean value) {}
+
+            @Override
+            public void validate(final Boolean value, final Map<Setting<?>, Object> settings) {
+                final Boolean clusterAwarenessSetting = (Boolean) settings.get(CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING);
+                if (clusterAwarenessSetting == false && value == true) {
+                    throw new IllegalArgumentException(
+                        "To enable "
+                            + USE_FORCE_ZONE_FOR_REPLICA_SETTING.getKey()
+                            + ", "
+                            + CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING.getKey()
+                            + " should be enabled"
+                    );
+                }
+            }
+
+            @Override
+            public Iterator<Setting<?>> settings() {
+                final List<Setting<?>> settings = Collections.singletonList(CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING);
+                return settings.iterator();
+            }
+        },
```

Comment:
we can skip this validation . Even if user sets it without `CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING`, it wouldn't matter .

Reply:
I think, we should have the validation since it will also help in avoid confusion for user in case they set this flag and don't see any change in the default replica count. Having this will make sure that User are aware of when this setting is applicable and can be used. @gbbafna